### PR TITLE
Add LOGIQ backend.

### PR DESCRIPTION
**Pull request by abhikhnvasara**
Add support for LOGIQ (https://logiq.ai/) backend.

### CODE REVIEW

File: tools/sigma/backends/logiq.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import re
+from .base import BaseBackend
+from .mixins import QuoteCharMixin
+import json
+
+class LogiqBackend(BaseBackend, QuoteCharMixin):
+    """Generates Perl compatible regular expressions and puts 'grep -P' around it"""
```
Comment (thomaspatzke) [inline tools/sigma/backends/logiq.py]: Description doesn't matches backends target name.

File: tools/sigma/backends/logiq.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        # print("generateORNode: ", node)
+        return "%s" % " || ".join([self.generateNode(val) for val in node])
+
+    def generateANDNode(self, node):
+        # print("generateORNode: ", node)
+        return "%s" % " && ".join([self.generateNode(val) for val in node])
+        
+    def generateNOTNode(self, node):
+        # print("generateNOTNode: ", node)        
+        return "%s" % self.generateNode(node.item)
```
Comment (thomaspatzke) [inline tools/sigma/backends/logiq.py]: This passes the value as-is, is this correct? Would expect a negation operator here.

File: tools/sigma/backends/logiq.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    def generateSubexpressionNode(self, node):
+        # print("generateSubexpressionNode: ", node)        
+        return "%s" % self.generateNode(node.items)
```
Comment (thomaspatzke) [inline tools/sigma/backends/logiq.py]: Would expect a grouping syntax here, parenthesis around the value etc.

File: tools/sigma/backends/logiq.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            return self.generateNode(value)
+
+    def generateValueNode(self, node):
+        # print("generateValueNode: ", node)
+        return "message =~ '" + self.cleanValue(str(node)).strip() + "'"
+
+    def generateNULLValueNode(self, node):
+        # print("generateNULLValueNode: ", node)
+        key, value = node
+        return "%s" % key
```
Comment (thomaspatzke) [inline tools/sigma/backends/logiq.py]: No comparison against a null value or similar?

**Comment by abhikhnvasara** [comment]
Thanks for reviewing and directing me to use SignleTextQueryBackend class. I made changes to use that and it made adding the backend trivial.

**Comment by abhikhnvasara** [comment]
Hi @thomaspatzke ,  Hope You are doing well. I have implemented the requested changes. When you get a chance could you please re-review?

**Comment by thomaspatzke** [comment]
Sorry for the delay, integrating it now!

**Comment by thomaspatzke** [comment]
Final CI test is running. Can you please check if the fixes I've introduced in commit 1c5c8047fdf87d75c35fda7364e455dd822c0428 are valid?

**Pull request merged**